### PR TITLE
passwordFields: add logic to cleanup secrets on update

### PR DESCRIPTION
**Pull request by kinarashah**
When password fields are not present in incoming data, need to delete secrets
from existing data. Earlier cleanup was done only on object deletion, not update.

### CODE REVIEW

**Comment by leodotcloud** [comment]
@kinarashah is there an existing issue? I do not know the context, trying to understand.

**Comment by kinarashah** [comment]
@leodotcloud Issue was brought up by @ aiwantaozi who is testing this for cluster logging, 

Context: 
Password store is used to fetch password type fields for given types and store their values into kubernetes secret before the actual store methods get called. 

Issue:
We also need to cleanup those existing secrets, which we only did on object delete. So when the object map was updated to None, or if it had nested arrays and one of the entries was removed, need to fetch secrets and delete them.

File: pkg/api/store/password/store.go [hunk 1]
```diff
@@ -189,6 +193,45 @@ func (p *PasswordStore) replacePasswords(sepData, data, existing map[string]inte
 					}
 				}
 			}
+			// delete remaining secrets from existingArr
+			for _, ind := range existingDataMap {
+				if err := p.replacePasswords(convert.ToMapInterface(sepVal), nil, existArr[ind]); err != nil {
```
Comment (mrajashree) [inline pkg/api/store/password/store.go]: The `existingDataMap` is called at this point left with existing keys that aren't present in the new data, so why do we have to call replacePasswords and not deleteSecrets?

**Comment by mrajashree** [comment]
LGTM

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/19053

**Pull request merged**